terminal.go

```diff
@@ -718,25 +718,58 @@ func (g *LightningTerminal) setUpLNDClients() error {
 		clientOptions = append(clientOptions, lndclient.Insecure())
 	}
 
+	// checkRunning checks if we should continue running for the duration of
+	// the defaultStartupTimeout, or else returns an error indicating why
+	// a shut down is needed.
+	checkRunning := func() error {
+		select {
+		case err := <-g.errQueue.ChanOut():
+			return fmt.Errorf("error from subsystem: %v", err)
+
+		case <-lndQuit:
+			return fmt.Errorf("LND has stopped")
+
+		case <-interceptor.ShutdownChannel():
+			return fmt.Errorf("received the shutdown signal")
+
+		case <-time.After(defaultStartupTimeout):
```

Comment:
If we want this to be the maximum time we wait in total, then we need to create the `time.After()` channel outside of the callback. Otherwise we wait this long for every re-try (since we create a new ticker channel each time).

Reply:
Or do we want to wait 5 seconds every time? Wasn't super clear to me from the comment above.

Reply:
> Or do we want to wait 5 seconds every time? Wasn't super clear to me from the comment above.

Yeah this was my intention, so that we don't spin as per @ellemouton's comment https://github.com/lightninglabs/lightning-terminal/pull/694#discussion_r1426517923, but still try to reinitiate the lndclient(s) on failure to set them up.